azkaban-common/src/main/java/azkaban/trigger/builtin/BasicTimeChecker.java

```diff
@@ -75,17 +87,33 @@ public ReadablePeriod getPeriod() {
   public long getNextCheckTime() {
     return nextCheckTime;
   }
+  public String getCronExpression() {
+    return cronExpression;
+  }
+
+  private void parseCronExpression() {
+    if (cronExpression != null && timezone != null) {
+      try {
+        cronExecutionTime = new CronExpression(cronExpression);
+        cronExecutionTime.setTimeZone(TimeZone.getTimeZone(timezone.getID()));
```

Comment:
Add a comment to explain that the the timezone types need to be converted? Thoughts on how to avoid this conversion in the future? What if we move to Java 8 only?